src/components/ControlBar.tsx

```diff
@@ -40,23 +43,75 @@ export const RestoreBtn = styled(Button)`
 `;
 
 interface ControlBarProps {
+  modelSaveSubs: Array<(save: ModelSave) => void>;
```

Comment:
As you are now using a callback to pass the data, it seems like the worker is not needed any more.